x/bank/keeper/keeper.go

```diff
@@ -26,27 +28,32 @@ type WrappedBankKeeper struct {
 
 	distrKeeper   types.DistributionKeeper
 	accountKeeper types.AccountKeeper
+	logger        log.Logger
 }
 
 // NewWrappedBankKeeperBurnToCommunityPool creates a new instance of WrappedBankKeeper
 // with its distrKeeper and accountKeeper members set to nil.
-func NewWrappedBankKeeperBurnToCommunityPool(bk bankkeeper.Keeper, acc types.AccountKeeper) WrappedBankKeeper {
-	return WrappedBankKeeper{bk, nil, acc}
+func NewWrappedBankKeeperBurnToCommunityPool(
+	bk bankkeeper.Keeper,
+	ak types.AccountKeeper,
+	logger log.Logger,
+) WrappedBankKeeper {
+	return WrappedBankKeeper{
+		Keeper:        bk,
+		distrKeeper:   nil,
+		accountKeeper: ak,
+		logger:        logger.With(log.ModuleKey, "x/wrappedbank"),
+	}
```

Comment:
There is an issue on this as MsgSend of bank module will try to cast keeper to `BaseKeeper` but it's the `WrappedBankKeeper`. This makes it causes tx to revert.

<img width="790" alt="image" src="https://github.com/user-attachments/assets/c931b0fe-1970-4ef8-9f9c-f3b1fa26b10e">

<img width="921" alt="image" src="https://github.com/user-attachments/assets/9762d507-d2c5-4c3f-802b-4844c1e41766">